File: 20 days/day2/f6.py
from flask import Flask, request, jsonify
from flask_cors import CORS

app = Flask(__name__)
CORS(app)

@app.route('/receive', methods=['POST'])
def receive_data():
    if request.is_json:
        data = request.get_json()
        print("Received data:", data)
        return jsonify({"message": "Data received", "data": data}), 200
    else:
        return jsonify({"error": "Invalid JSON"}), 400



if __name__ == '__main__':
    app.run(host='127.0.0.1', port=5000)

File: 20 days/day2/f6.js
const express = require('express');
const axios = require('axios');
const app = express();
const port = 3000;


app.use(express.json());

app.post('/send-to-flask',async(req,res) =>{
    const payload ={
        name:'Francis',
        role:'Developer'
    };

    try{
        const respose = await axios.post('http://127.0.0.1:5000/recieve',payload)
        res.json(respose.data);
    }catch(error){
        console.error('Error',error.message);
        res.status(500).send('Failed to send data to flask');
    }
});

app.listen(poirt,() =>{
    console.log(`Express running at http://localhost:${port}`);
}) 
